src/archive.cpp

```diff
@@ -86,6 +86,96 @@ namespace zim
     return ret;
   }
 
+  zim::FileImpl::FindxResult findFavicon(FileImpl& impl)
+  {
+    for(auto ns:{'-', 'I'}) {
+      for (auto& path:{"favicon", "favicon.png"}) {
+        auto r = impl.findx(ns, path);
+        if (r.first) {
+          return r;
+        }
+      }
+    }
+    throw EntryNotFound("No favicon found.");
+  }
+
+  unsigned int parseIllustrationPathToSize(const std::string& str) {
+    constexpr auto illuSize = 13; // strlen("Illustration_");
+    auto xpos = str.find('x', illuSize);
+    if (xpos == std::string::npos) {
+      throw std::runtime_error("");
+    }
+    auto atpos = str.find("@1", xpos);
+    if (atpos == std::string::npos) {
+      throw std::runtime_error("");
+    }
+    if (str.substr(illuSize, xpos-illuSize) != str.substr(xpos+1, atpos-xpos-1)) {
+      throw std::runtime_error("");
+    }
+    std::istringstream ss(str.substr(illuSize, xpos-illuSize));
+    unsigned int ret;
+    ss >> ret;
+    return ret;
+  }
```

Comment:
This would accept `"Illustration_64x64@1.5"` too (and, as far as I know, non-integer scale values are allowed).

Using good old `sscanf()` is justified here:

```c++
unsigned int parseIllustrationPathToSize(const std::string& s) {
    int n;
    unsigned int w, h;
    if ( sscanf(s.c_str(), "Illustration_%ux%u@1%n)", &w, &h, &n) == 2
         && n == s.size() && w == h) {
        return w;
    }

    throw std::runtime_error("");
}
```

The only problem with this version is that it ignores any whitespace before the size (e.g. `"Illustration_    64x 64"` is happily accepted). A similar issue is also present in your version, however only if both width and height values are preceded with the same amount of whitespace (and it can be fixed easier in your implementation).

Reply:
Done, based on your implementation (in several commits).